db/db_impl_write.cc

```diff
@@ -534,23 +566,76 @@ Status DBImpl::PipelinedWriteImpl(const WriteOptions& write_options,
   return w.FinalStatus();
 }
 
+Status DBImpl::UnorderedWriteMemtable(const WriteOptions& write_options,
+                                      WriteBatch* my_batch,
+                                      WriteCallback* callback, uint64_t log_ref,
+                                      SequenceNumber seq) {
+  PERF_TIMER_GUARD(write_pre_and_post_process_time);
+  StopWatch write_sw(env_, immutable_db_options_.statistics.get(), DB_WRITE);
+
+  WriteThread::Writer w(write_options, my_batch, callback, log_ref,
+                        false /*disable_memtable*/);
+
+  size_t total_count = 0;
+  size_t total_byte_size = 0;
+
+  if (w.CheckCallback(this)) {
+    if (w.ShouldWriteToMemtable()) {
+      w.sequence = seq;
+      total_count += WriteBatchInternal::Count(my_batch);
+    }
+    total_byte_size = WriteBatchInternal::AppendedByteSize(
+        total_byte_size, WriteBatchInternal::ByteSize(w.batch));
+  }
+
+  auto stats = default_cf_internal_stats_;
+  stats->AddDBStats(InternalStats::NUMBER_KEYS_WRITTEN, total_count);
+  RecordTick(stats_, NUMBER_KEYS_WRITTEN, total_count);
+  stats->AddDBStats(InternalStats::BYTES_WRITTEN, total_byte_size);
+  RecordTick(stats_, BYTES_WRITTEN, total_byte_size);
+  RecordInHistogram(stats_, BYTES_PER_WRITE, total_byte_size);
+
+  if (!w.CallbackFailed()) {
+    WriteStatusCheck(w.status);
+  }
+
+  ColumnFamilyMemTablesImpl column_family_memtables(
+      versions_->GetColumnFamilySet());
+  w.status = WriteBatchInternal::InsertInto(
+      &w, w.sequence, &column_family_memtables, &flush_scheduler_,
+      write_options.ignore_missing_column_families, 0 /*log_number*/, this,
+      true /*concurrent_memtable_writes*/);
+  auto pending_cnt = pending_memtable_writes_.fetch_sub(1) - 1;
+  if (pending_cnt == 0) {
+    switch_cv_.notify_all();
+  }
+
+  if (!w.FinalStatus().ok()) {
+    return w.FinalStatus();
+  }
+  return Status::OK();
+}
+
 // The 2nd write queue. If enabled it will be used only for WAL-only writes.
 // This is the only queue that updates LastPublishedSequence which is only
 // applicable in a two-queue setting.
-Status DBImpl::WriteImplWALOnly(const WriteOptions& write_options,
+Status DBImpl::WriteImplWALOnly(WriteThread& write_thread,
```

Comment:
I commented previously that I think the function should be renamed. It's not for write request with WAL only anymore, but more broadly used.

Reply:
Maybe after you update the comments from line 622 to 624, a better name will come out. Right now, I'm not very clear what you want the function to be, though I feel like the name now is not accurate enough.

Reply:
Last year we went back and forth on the name multiple times. Please do tell me your suggestion and I will replace the name accordingly.